elementary/clients/api/api.py

```diff
@@ -1,6 +1,16 @@
+from collections import defaultdict
+from typing import Any
+
 from elementary.clients.dbt.dbt_runner import DbtRunner
 
 
 class APIClient:
     def __init__(self, dbt_runner: DbtRunner):
         self.dbt_runner = dbt_runner
+        self.run_cache = defaultdict(lambda: None)
```

Comment:
Added a run cache - now we want have to pass all the API results for each API call that is relying on it, and we can make a mistakable second operation run to not be executed :)